sql/parser/type_check.go

```diff
@@ -228,6 +231,63 @@ func typeCheckComparisonOp(op ComparisonOp, dummyLeft, dummyRight Datum) (Datum,
 		dummyLeft.Type(), op, dummyRight.Type())
 }
 
+func typeCheckConditionalExpr(expr *ConditionalExpr) (Datum, error) {
+	switch expr.Operator {
+	case If:
+		if len(expr.Exprs) != 3 {
```

Comment:
> Would an IfExpr that implements ConditionalExpr only have slots for 3 expressions?

Yeah, that's what I'm suggesting.

> Then we would need different code in WalkExpr.

I think that's OK, it wouldn't be much code.

If you made ConditionalExpr an interface, you could pull CaseExpr into that interface as well, which may allow you to share the typeCheck/evalExpr implementations